jx/build/JXML.js

```diff
@@ -186,10 +186,10 @@ JXML.prototype.generateJS = function(build_assets) {
  * TODO: text interleaved with elements not handled well
  */
 function XMLToJSON(element) {
-	var name = (element.namespaceURI? element.namespaceURI + '/' : '') + element.tagName,
+	var name = (element.namespaceURI? element.namespaceURI + '/' : '') + element.localName,
 		children = {},
 		childNodes = element.childNodes,
-		children_index = 1,
+		children_index = 0,
```

Comment:
actually we thought wrong this afternoon. There were no problem with this code :p
``` javascript
a = 1; 
console.log(a++)  // 1
console.log(a)  // 2
```